bin/wasm-node/rust/src/network_service.rs

```diff
@@ -235,6 +236,21 @@ impl NetworkService {
         result
     }
 
+    /// Sends transaction to the given peer.
+    // TODO: more docs
+    pub async fn announce_transaction(
+        self: Arc<Self>,
+        transaction: &[u8],
+    ) -> Result<(), QueueNotificationError> {
+        for target in self.peers_list().await {
+            let _ = self
+                .network
+                .announce_transaction(&target, 0, &transaction)
+                .await?;
```

Comment:
```suggestion
                .await;
```

I didn't catch this!

Errors are normal here. We don't want to make it look like we have failed to announce a transaction when it's just one peer that hasn't received it.

Please make the function return an error only if we couldn't send the transaction to anyone, or if the list of peers was empty.

Reply:
And mention this behaviour in the documentation.